blockchain/state/statedb.go

```diff
@@ -603,7 +662,15 @@ func (self *StateDB) GetOrNewSmartContract(addr common.Address) *stateObject {
 // createObject creates a new state object. If there is an existing account with
 // the given address, it is overwritten and returned as the second return value.
 func (self *StateDB) createObject(addr common.Address) (newobj, prev *stateObject) {
-	prev = self.getStateObject(addr)
+	prev = self.getDeletedStateObject(addr) // Note, prev might have been deleted, we need that!
```

Comment:
It seems that this code changes the semantics of the previous logic. Is it okay?
In the previous code, `prev` becomes `nil` when it was deleted. And, the journal will append `createObjectChange` in the following logic.
With your change, `prev` becomes the old `obj` when it was deleted before. And, the journal will append `resetObjectChange` in the follwoing logic.